h2/src/main/org/h2/mode/PgCatalogTable.java

```diff
@@ -351,6 +368,34 @@ public PgCatalogTable(Schema schema, int id, int type) {
             break;
         case PG_AUTHID:
             break;
+        case PG_CONSTRAINT:
+            for (SchemaObject obj : database.getAllSchemaObjects(DbObject.CONSTRAINT)) {
+                Constraint constraint = (Constraint) obj;
+                Constraint.Type constraintType = constraint.getConstraintType();
+                if (constraintType == Constraint.Type.DOMAIN) {
+                    continue;
+                }
+                Table table = constraint.getTable();
+                if (hideTable(table, session)) {
+                    continue;
+                }
+                Table refTable = constraint.getRefTable();
+                add(session,
+                        rows,
+                        // OID
+                        ValueInteger.get(constraint.getId()),
+                        // CONNAME
+                        constraint.getName(),
+                        // CONTYPE
+                        StringUtils.toLowerEnglish(constraintType.toString().substring(0, 1)),
```

Comment:
`constraintType.getSqlName()` is more reliable than `toString()`.

Reply:
Yes, fixed. Foreign key should return `f` but not `r`.